pkg/server/status/recorder.go

```diff
@@ -486,22 +486,42 @@ func (mr *MetricsRecorder) GetMetricsMetadata(
 	mr.mu.logRegistry.WriteMetricsMetadata(srvMetrics)
 	mr.mu.sysRegistry.WriteMetricsMetadata(srvMetrics)
 
-	// Get a random storeID.
-	var sID roachpb.StoreID
+	mr.getStoreMetricsMetadata(nodeMetrics)
+	return nodeMetrics, appMetrics, srvMetrics
+}
+
+// GetRecordedMetricNames takes a map of metric metadata and returns a map
+// of the metadata name to the name the metric is recorded with in tsdb.
+func (mr *MetricsRecorder) GetRecordedMetricNames(
+	allMetadata map[string]metric.Metadata,
+) map[string]string {
+	storeMetricsMap := make(map[string]metric.Metadata)
+	tsDbMetricNames := make(map[string]string, len(allMetadata))
+	mr.getStoreMetricsMetadata(storeMetricsMap)
+	for k := range allMetadata {
+		prefix := nodeTimeSeriesPrefix
+		if _, ok := storeMetricsMap[k]; ok {
+			prefix = storeTimeSeriesPrefix
+		}
 
-	storeFound := false
-	for storeID := range mr.mu.storeRegistries {
-		sID = storeID
-		storeFound = true
-		break
+		tsDbMetricNames[k] = fmt.Sprintf(prefix, k)
 	}
+	return tsDbMetricNames
+}
 
-	// Get metric metadata from that store because all stores have the same metadata.
-	if storeFound {
-		mr.mu.storeRegistries[sID].WriteMetricsMetadata(nodeMetrics)
+// getStoreMetricsMetadata Gets a store from mr.mu.storeRegistries and writes
+// the metrics metadata to the provided map.
+func (mr *MetricsRecorder) getStoreMetricsMetadata(metricsMetadata map[string]metric.Metadata) {
+	if len(mr.mu.storeRegistries) == 0 {
+		return
 	}
 
-	return nodeMetrics, appMetrics, srvMetrics
+	// All store registries should have the same metadata, so only the metadata
+	// from the first store is used to write to metricsMetadata.
+	for _, registry := range mr.mu.storeRegistries {
+		registry.WriteMetricsMetadata(metricsMetadata)
```

Comment:
Suggestion, but it might be clearer to do something like

```go
registry := mr.mu.storeRegistries[0]
registry.WriteMetricsMetadata(metricsMetadata)
```

Reply:
`mr.mu.storeRegistries` is a map, not a slice, so this is effectively grabbing a random store registry and writing them to metrics metadata.

Reply:
oh fascinating, got it thanks!